scripts/hardhat-tasks-functions.js

```diff
@@ -0,0 +1,144 @@
+/* eslint-disable no-undef */ // This is to disable warning for hre undefined in file
+/* eslint-disable import/no-extraneous-dependencies */ // This is to disable warnings for imports
+const fs = require('fs-extra');
+const { exec } = require('child_process');
+const axios = require('axios');
+const path = require('path');
+
+const hardhatImport = import('../hardhat.config.js');
+
+function execShellCommand(cmd) {
+    return new Promise((resolve) => {
+        exec(cmd, (error, stdout, stderr) => {
+            if (error) {
+                console.warn(error);
+            }
+            resolve(stdout || stderr);
+        });
+    });
+}
+function sleep(ms) {
+    return new Promise((resolve) => {
+        setTimeout(resolve, ms);
+    });
+}
+async function deployContract(contractName, gasPriceSelected) {
+    await execShellCommand('npx hardhat compile');
+    const overrides = {
+        // The price (in wei) per unit of gas
+        gasPrice: hre.ethers.utils.parseUnits(gasPriceSelected, 'gwei'),
+    };
+    const [deployer] = await hre.ethers.getSigners();
+    console.log('Deploying contracts with the account:', deployer.address);
+
+    console.log('Account balance:', (await deployer.getBalance()).toString());
+    const Contract = await hre.ethers.getContractFactory(
+        `contracts/flattened/${contractName}.sol:${contractName}`,
+    );
+    const contract = await Contract.deploy(overrides);
```

Comment:
Can we add a confirm yes/no prompt and info (what we are deploying, what network, how much it will cost)